lib/rex/post/smb/ui/console/command_dispatcher/shares.rb

```diff
@@ -318,7 +344,198 @@ def cmd_cat(*args)
           def cmd_cd_tabs(_str, words)
             return [] if words.length > 1
 
-            @@cat_opts.option_keys
+            @@cd_opts.option_keys
+          end
+
+          def cmd_upload(*args)
+            if args.include?('-h') || args.include?('--help')
+              cmd_upload_help
+              return
+            end
+            local_path = nil
+            remote_path = nil
+
+            @@upload_opts.parse(args) do |_opt, idx, val|
+              case idx
+              when 0
+                local_path = val
+              when 1
+                remote_path = val
+              else
+                print_warning('Too many parameters')
+                cmd_upload_help
+              end
+            end
+
+            if local_path.blank?
+              print_error('No local path given')
+              return
+            end
+
+            remote_path = Rex::Post::File.basename(local_path) if remote_path.nil?
+            full_path = as_ntpath(Pathname.new(shell.cwd).join(remote_path).to_s)
+
+            fd = simple_client.open(full_path, 'wct', write: true)
+
+            data = ::File.read(local_path, ::File.size(local_path), mode: 'rb')
```

Comment:
```suggestion
            data = ::File.binread(local_path)
```

Reply:
We should probably try reading the file first, so if it explodes locally - we haven't tried opening up remote connections yet - and it'll be les noise-y in the user error scenario